onedal/linear_model/linear_model.cpp

```diff
@@ -80,75 +78,108 @@ auto get_onedal_result_options(const py::dict& params) {
-                                             linear_regression::task::regression>(intercept);
-    }
-};
-
 struct params2desc {
     template <typename Float, typename Method, typename Task>
     auto operator()(const py::dict& params) {
         using namespace dal::linear_regression;
 
         const auto intercept = params["intercept"].cast<bool>();
 
-        auto desc = descriptor_creator<Float, Method, Task>::get(intercept).set_result_options(
-            get_onedal_result_options(params));
+        auto desc = linear_regression::descriptor<Float, Method, Task>(intercept)
+            .set_result_options(get_onedal_result_options(params));
         return desc;
     }
 };
 
 template <typename Policy, typename Task>
-struct init_train_ops_dispatcher {};
-
-template <typename Policy>
-struct init_train_ops_dispatcher<Policy, linear_regression::task::regression> {
-    void operator()(py::module_& m) {
-        using Task = linear_regression::task::regression;
-
+void init_train_ops(py::module& m) {
 #if defined(ONEDAL_VERSION) && ONEDAL_VERSION >= 20240000
-        using train_hyperparams_t = dal::linear_regression::detail::train_parameters<Task>;
-        m.def("train", [](
-            const Policy& policy,
-            const py::dict& params,
-            const train_hyperparams_t& hyperparams,
-            const table& data,
-            const table& responses) {
-                using namespace dal::linear_regression;
-                using input_t = train_input<Task>;
-                train_ops_with_hyperparams ops(
-                    policy, input_t{ data, responses }, params2desc{}, hyperparams);
-                return fptype2t{ method2t{ Task{}, ops } }(params);
-            }
-        );
+    using train_hyperparams_t = dal::linear_regression::detail::train_parameters<Task>;
+    m.def("train", [](
+        const Policy& policy,
+        const py::dict& params,
+        const train_hyperparams_t& hyperparams,
+        const table& data,
+        const table& responses) {
+            using namespace dal::linear_regression;
+            using input_t = train_input<Task>;
+            train_ops_with_hyperparams ops(
+                policy, input_t{ data, responses }, params2desc{}, hyperparams);
+            return fptype2t{ method2t{ Task{}, ops } }(params);
+        }
+    );
 #endif // defined(ONEDAL_VERSION) && ONEDAL_VERSION >= 20240000
-        m.def("train", [](
-            const Policy& policy,
-            const py::dict& params,
-            const table& data,
-            const table& responses) {
-                using namespace dal::linear_regression;
-                using input_t = train_input<Task>;
-                train_ops ops(policy, input_t{ data, responses }, params2desc{});
-                return fptype2t{ method2t{ Task{}, ops } }(params);
-            }
-        );
-    }
+    m.def("train", [](
+        const Policy& policy,

```

Comment:
I think here should be `else` in case of onedal version < 20240000

Reply:
I guess it was done like that to keep ability to call methods without hyperparameters from C++ side. probably they act differently. here it is done exactly like in all other algorithms with hyperparameters. I think if we want to change such behavior then it is better to do in separate PR and for all algos.